pages/index.tsx

```diff
@@ -105,8 +108,19 @@ const Container: React.FC<{ children: React.ReactNode }> = ({ children }) => (
   <div className="max-w-[600px] mx-5 sm:mx-auto">{children}</div>
 );
 
-const fetchedAtTimeTitle = (t: string) =>
-  DateTime.fromISO(t).toLocaleString(DateTime.DATETIME_FULL);
+const useTimeFormatter = () => {
+  // Set formatter for server - simply return a string
+  const [format, setFormat] = useState(() => (a: string) => "");
+
+  // Set formatter on client - will run in client TZ
+  useEffect(() => {
+    const clientFormatter = (t: string) =>
+      DateTime.fromISO(t).toFormat("dd LLL yyyy h:mm a");
+    setFormat(() => clientFormatter);
+  }, []);
+
+  return format;
```

Comment:
The key part of the useEffect is that it doesn't "depend" on anything - you can see in the dep arr:
```
 }, []);
```

So it will fire just that one time - it doesn't depend on the state, so doesn't fire when the state changes